blog/_posts/2023-04-19-conda-r-geospatial.md

```diff
@@ -0,0 +1,87 @@
+
+To create an environment we use the `conda create` command. In the snippet below we specify the environments name, some packages we want to install with specific versions and the [Conda channel](https://docs.conda.io/projects/conda/en/stable/user-guide/concepts/channels.html) from which we want to install these packages from.
+
+```bash
+# create a conda environment called r-geo
+# and install 3 packages from the conda forge channel
+conda create -n r-geo -c conda-forge r-base=4.1.1 gdal=3 udunits2
+```
+
+Now this step above is sort of cheating. I only specify the geospatial library `gdal` here but Conda also installs an additional series of dependencies for those libraries.
+So I actually get a number of other critical geospatial libraries too such as [PROJ](https://proj.org/) and [GEOS](https://libgeos.org/).
+This is an important caveat when using Conda, is that it doesn't just install the packages we ask for but also installs any dependencies the packages we request depend on.
+To be fully reproducible across different systems we need to be aware of this and conscious of how we document these steps.
+Conda provides a [number of commands for outputting a specification file](https://arctraining.github.io/hpc2-software/course/conda.html#sharing-conda-environments) for your environment that you can shared and have others reproduce your environment from. 
+We **strongly** recommend that if you're using Conda you create these environment specification files and include them in your project repositories to help reproduce your environment. 
+However that is a topic outside of the scope of this particular blog post so lets now get back to installing `terra`.
+
+With our environment created we now need to enter that environment with our shell. 
+Conda environments are these isolated bubbles which we need to step into and out of in the shell in order for that isolation to work.
+To enter a Conda environment and have access to all the packages we've installed in it we use the command `conda activate ENV_NAME`, where `ENV_NAME` is the name of the environment. 
+However, on ARC4 we use the older version of this command `source activate ENV_NAME` to avoid some niggles we have with the newer behaviour of `conda activate`.
+
+```bash
+# activate the environment to use the packages you've installed
+source activate r-geo
+```
+
+After running `source activate` our prompt will be prepended with the name of our environment and we can now access the packages we installed into this environment.
+To install our R package we can either start up the R console and run `install.packages` directly or use the `R -e` command to pass in a string to the R to be run without opening a full R console.
+
+```bash
+# Install packages in R without starting an R console
+(r-geo) R -e 'install.packages("terra");' 
```

Comment:
I have no idea how to install anything with R. So I followed all your instructions and got the following error:

```bash
(r-geo) [earpte@login2.arc4 ~]$ R -e 'install.packages("terra");'

R version 4.1.1 (2021-08-10) -- "Kick Things"
Copyright (C) 2021 The R Foundation for Statistical Computing
Platform: x86_64-conda-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> install.packages("terra");
Error in contrib.url(repos, type) : 
  trying to use CRAN without setting a mirror
Calls: install.packages -> startsWith -> contrib.url
Execution halted
```

Reply:
```suggestion
(r-geo) R -e 'install.packages("terra", repo="https://cloud.r-project.org/");' 
```